jscomp/main/jsoo_refmt_main.ml

```diff
@@ -0,0 +1,251 @@
+(* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
+
+(**
+`jsoo_refmt_main` is the JSOO compilation entry point for building BuckleScript + Refmt as one bundle.
+This is usually the file you want to build for the full playground experience.
+*)
+module Js = struct
+  module Unsafe = struct
+    type any
+    external inject : 'a -> any = "%identity"
+    external get : 'a -> 'b -> 'c = "caml_js_get"
+    external set : 'a -> 'b -> 'c -> unit = "caml_js_set"
+    external pure_js_expr : string -> 'a = "caml_pure_js_expr"
+    let global = pure_js_expr "joo_global_object"
+    type obj
+    external obj : (string * any) array -> obj = "caml_js_object"
+  end
+  type (-'a, +'b) meth_callback
+  type 'a callback = (unit, 'a) meth_callback
+  external wrap_callback : ('a -> 'b) -> ('c, 'a -> 'b) meth_callback = "caml_js_wrap_callback"
+  external wrap_meth_callback : ('a -> 'b) -> ('a, 'b) meth_callback = "caml_js_wrap_meth_callback"
+  type + 'a t
+  type js_string
+  external string : string -> js_string t = "caml_js_from_string"
+  external to_string : js_string t -> string = "caml_js_to_string"
+  external create_file : js_string t -> js_string t -> unit = "caml_create_file"
```

Comment:
can this module be shared by making a new file called js_ffi_unsafe?

Reply:
I created a file `jsoo_common.ml` to share the `Js` namespace and the `mk_js_error` function etc.